notebooks/Functions_EDA_CMM.py

```diff
@@ -0,0 +1,40 @@
+import pandas as pd
+import numpy as np
+
+
+# Define function for generating dataframe which calculates basketsizes per order.
+
+def calc_basketsize (purchases):
+    """Function to generate dataframe with basketsizes out of dataframe with single purchases (e.g. dataframe from transaction_train.csv). 
+    Assumption: Purchases of an individual customer on one day form an order.
+
+    Args:
+        purchases (_dataframe_): Dataframe which contains single purchases per customer in each row
+    """    
+    
+    purchases['datetime'] = pd.to_datetime(purchases['t_dat'])
+
+    orderbaskets = purchases.groupby(['datetime', 'customer_id']).size().reset_index()
```

Comment:
Um die "macht" von groupby zu erweitern gibt es noch die [agg](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.agg.html) funktion.
  Man könnte auch überlegen eine [pivot-Tabelle zu nutzen](https://towardsdatascience.com/pandas-pivot-the-ultimate-guide-5c693e0771f3). [to_pivot doc](https://pandas.pydata.org/docs/reference/api/pandas.pivot_table.html)

Reply:
Guter Tipp, checke später, wie man die agg-funktion einbauen kann und ob das gleiche rauskommt.